cpp/arcticdb/python/python_module.cpp

```diff
@@ -227,12 +191,33 @@ void register_error_code_ecosystem(py::module& m) {
     setattr(m, "enum_value_to_prefix", enum_value_to_prefix);
     m.def("get_error_category", &get_error_category);
 
-    py::register_exception<InternalException>(m, "InternalException");
-    py::register_exception<NormalizationException>(m, "NormalizationException");
-    py::register_exception<MissingDataException>(m, "MissingDataException");
-    py::register_exception<SchemaException>(m, "SchemaException");
-    py::register_exception<StorageException>(m, "StorageException");
-    py::register_exception<SortingException>(m, "SortingException");
+    // legacy exception base type kept for backwards compat with Man Python client
+    struct ArcticCompatibilityException : public std::runtime_error {};
```

Comment:
Not a big fan of creating a class inside a function.

The C++ exceptions can benefit from a common, non-templated base class as well.

Also, don't use `register_exception` unless you want to see the exception type thrown in C++  as well.

Reply:
I think the use of `register_exception` is correct here. https://pybind11.readthedocs.io/en/stable/advanced/exceptions.html

Reply:
That should never happen. It's by design to keep this here in the function and hide it as much as possible.